core/src/test/java/com/novoda/noplayer/player/PlayerFactoryTest.java

```diff
@@ -0,0 +1,151 @@
+package com.novoda.noplayer.player;
+
+import android.content.Context;
+
+import com.novoda.noplayer.Player;
+import com.novoda.noplayer.drm.DownloadedModularDrm;
+import com.novoda.noplayer.drm.DrmHandler;
+import com.novoda.noplayer.drm.DrmType;
+import com.novoda.noplayer.drm.StreamingModularDrm;
+import com.novoda.noplayer.exoplayer.DrmSessionCreator;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static com.novoda.noplayer.player.PrioritizedPlayerTypes.prioritizeExoPlayer;
+import static com.novoda.noplayer.player.PrioritizedPlayerTypes.prioritizeMediaPlayer;
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+
+@RunWith(Enclosed.class)
+public class PlayerFactoryTest {
+
+    public static abstract class Base {
+
+        static final StreamingModularDrm STREAMING_MODULAR_DRM = mock(StreamingModularDrm.class);
+        static final DownloadedModularDrm DOWNLOADED_MODULAR_DRM = mock(DownloadedModularDrm.class);
+        static final Player EXO_PLAYER = mock(Player.class);
+        static final Player MEDIA_PLAYER = mock(Player.class);
+
+        @Rule
+        public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+        @Mock
+        Context context;
+
+        @Mock
+        PlayerFactory.ExoPlayerCreator exoPlayerCreator;
+
+        @Mock
+        PlayerFactory.MediaPlayerCreator mediaPlayerCreator;
+
+        PlayerFactory playerFactory;
+
+        @Before
+        public void setUp() {
+            given(exoPlayerCreator.createExoPlayer(any(Context.class), any(DrmSessionCreator.class))).willReturn(EXO_PLAYER);
+            given(mediaPlayerCreator.createMediaPlayer(any(Context.class))).willReturn(MEDIA_PLAYER);
+            playerFactory = new PlayerFactory(context, prioritizedPlayerTypes(), exoPlayerCreator, mediaPlayerCreator);
+        }
+
+        abstract PrioritizedPlayerTypes prioritizedPlayerTypes();
+
+    }
+
+    public static class GivenMediaPlayerPrioritized extends Base {
+
+        @Override
+        PrioritizedPlayerTypes prioritizedPlayerTypes() {
+            return prioritizeMediaPlayer();
+        }
+
+        @Test
+        public void givenMediaPlayerPrioritized_whenCreatingPlayer_thenReturnsMediaPlayer() {
```

Comment:
commit bc50c1a06def1fd7e4780f05470c3bf66cae45f7
Author: Ataul Munim <hello@ataulm.com>
Date:   Mon Jun 5 11:36:13 2017 +0100

    Remove test prefixes which are covered in test class name